itest/rust/src/property_test.rs

```diff
@@ -289,3 +290,28 @@ struct CheckAllExports {
 
 #[godot_api]
 impl CheckAllExports {}
+
+#[repr(i64)]
+#[derive(Property, Debug, PartialEq, Eq)]
+pub enum Test {
+    A = 0,
+    B = 1,
+    C = 2,
+}
+
+#[derive(GodotClass)]
+pub struct DeriveProperty {
+    #[var]
+    pub foo: Test,
+}
+
+#[godot_api]
+impl DeriveProperty {}
+
+#[itest]
+fn derive_property() {
+    let mut class = DeriveProperty { foo: Test::B };
+    assert_eq!(class.get_foo(), Test::B as i64);
+    class.set_foo(Test::C as i64);
+    assert_eq!(class.foo, Test::C);
```

Comment:
I wonder if it's possible that `get_foo/set_foo` return/accept directly the enum in Rust, not the `i64`?

Although maybe that's more complex; in that case it could be done later as a QoL improvement.

Reply:
The scope of this PR is more to add the proc macros than to modify library code - I'm not saying I won't be doing this or that it's completely outside the scope of this PR, but this is definitely not a priority for me rn.